staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1939,6 +1939,68 @@ type ClusterTrustBundleProjection struct {
 	Path string `json:"path" protobuf:"bytes,4,rep,name=path"`
 }
 
+// PodCertificateProjection provides a private key and X.509 certificate in the
+// pod filesystem.
+type PodCertificateProjection struct {
+	// Kubelet's generated CSRs will be addressed to this signer.
+	SignerName string `json:"signerName,omitempty" protobuf:"bytes,1,rep,name=signerName"`
+
+	// The type of keypair Kubelet will generate for the pod.
+	//
+	// Valid values are "RSA3072", "RSA4096", "ECDSAP256", "ECDSAP384", and
+	// "ED25519".  If left empty, Kubelet defaults to "ECDSAP256".
+	KeyType string `json:"keyType,omitempty" protobuf:"bytes,2,rep,name=keyType"`
+
+	// What is the maximum lifetime of the cert that should be issued?  The
+	// actual issued certificate may have any lifetime *less* than
+	// maxExpirationSeconds.
+	//
+	// This value is copied into the PodCertificateRequest that kubelet creates
+	// to request the certificate.
+	//
+	// If omitted or left at 0, kube-apiserver will set a default value of 86400
+	// (1 day).
+	//
+	// kube-apiserver can additionally override this with a shorter value based
+	// on a per-signer configuration, and it will reject values less than 3600
+	// (1 hour).
+	//
+	// The signer implementation is then free to issue a certificate with any
+	// lifetime *less* than maxExpirationSeconds.
+	MaxExpirationSeconds int32 `json:"maxExpirationSeconds,omitempty" protobuf:"varint,3,opt,name=maxExpirationSeconds"`
```

Comment:
prefer nil / absent to `0` for defaulting

```suggestion
	MaxExpirationSeconds *int32 `json:"maxExpirationSeconds,omitempty" protobuf:"varint,3,opt,name=maxExpirationSeconds"`
```

Reply:
Don.e